### PR TITLE
Enable zlib implicitly in cmake

### PR DESCRIPTION
**Pull request by willson-chen**
There is not option named `ENABLE_ZLIB` in `CMakeLists.txt`.

### CODE REVIEW

**Comment by willson-chen** [comment]
@vapier hi,
I thought `zlib` is mandatory for libgd. So it is only used for libpng, right? I will add the `ENABLE_ZLIB` option to CMakeLists.txt soon.

BTW, your new avatar is so cool. Do you draw it yourself? :)

**Comment by vapier** [comment]
we check for zlib support in a bunch of places in the code.  afaik, it's still working. it is often with png, but iirc, also for other things.

everyone has zlib, but the fact we can build gd without any external deps has a bit of a charm to it.

**Comment by willson-chen** [comment]
@vapier hi, I have a question. zlib is only used in png(libpng) and gd2 formate, and they have their own option(`ENABLE_PNG` and `ENABLE_GD_FORMAT`). IMHO, it is more reasonable that zlib is enabled implicitly when `ENABLE_PNG` or `ENABLE_GD_FORMAT` is **ON**.

**Comment by willson-chen** [comment]
Hi @vapier @cmb. Do you mind continuing this discussion when you are free?

I wanna close, improve and merge some open PR recently. But I need your help. :)

**Comment by vapier** [comment]
i'd be fine removing zlib as a dedicated knob for users to control iif we still only probe/link against it when other features require it

File: CMakeLists.txt [hunk 1]
```diff
@@ -100,16 +100,19 @@ else (USE_EXT_GD)
 	include(AC_HEADER_STDC)
 	include(CheckPrototypeExists)
 
+	if (ENABLE_GD_FORMATS)
+		FIND_PACKAGE(ZLIB)
```
Comment (vapier) [inline CMakeLists.txt:104]: you want `REQUIRED` here (and below) so we abort if it's not found

we prob should be doing that for all FIND_PACKAGE calls actually (although you don't have to fix it here)

Comment (willson-chen) [reply CMakeLists.txt:104]: yeah, we have another PR #472 to add `REQUIRED`.

**Comment by willson-chen** [comment]
> if we're collapsing this setting into GD2/PNG, we can scrub the codebase of all dedicated ZLIB refs.

Sorry, I can not fully understand the meaning. Could you please specify it?

**Comment by vapier** [comment]
change all the `FIND_PACKAGE(ZLIB)` to `FIND_PACKAGE(ZLIB REQUIRED)` and i'm fine merging this PR

**Comment by willson-chen** [comment]
> change all the `FIND_PACKAGE(ZLIB)` to `FIND_PACKAGE(ZLIB REQUIRED)` and i'm fine merging this PR

Done :)

**Comment by vapier** [comment]
thx bud

**Pull request merged**